portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,14 +33,93 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json;");
-    // response.getWriter().println("Hello Chisom!");
-    String jsonComments = "{\"Femi\": \"Nice page Chisom\", \"Dera\": \"Try to make the web page reponsive\", \"Kamsi\": \"Kudos\"}";
-    response.getWriter().println(jsonComments);
-  }
+   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Query query = new Query("UserData");
+      ArrayList<UserData> data = new ArrayList<UserData>();	
+      PreparedQuery results = datastore.prepare(query);
+
+      Date currentDate = new Date();  
+    
+      for (Entity entity : results.asIterable()) { 
+        long id = entity.getKey().getId();
+        String name = (String) entity.getProperty("name");
+        String email = (String) entity.getProperty("email");
+        String comment = (String) entity.getProperty("comment");
+        Date date = (Date) entity.getProperty("date");
+        UserData newData = new UserData(id, name, email, comment, date);
+        data.add(newData);
+        }
+
+        response.setContentType("application/json;");
+
+        Gson gson = new Gson();
+        final int numCommentsUserSelects = getNumCommentsUserSelects(request, "numValue", 5);
+        
+        final int numCommentsDisplayed = Math.min(numCommentsUserSelects, data.size());
+	    response.getWriter().println(gson.toJson(data.subList(0, numCommentsDisplayed)));
```

Comment:
nit: indentation is still off across many files - please fix

Reply:
done.